app/src/main/java/com/github/epfl/meili/posts/PostActivity.kt

```diff
@@ -90,15 +117,42 @@ class PostActivity : AppCompatActivity() {
 
         viewModel.initDatabase(FirestoreDatabase("forum/$postId/comments", Comment::class.java))
         viewModel.getElements().observe(this, { map ->
-            recyclerAdapter.submitList(map.toList())
-            recyclerAdapter.notifyDataSetChanged()
+            commentListener(map)
         })
     }
 
+    private fun commentListener(commentsMap: Map<String, Comment>) {
+        val newUsers = ArrayList<String>()
+        for ((commentId, comment) in commentsMap) {
+            newUsers.add(comment.authorUid)
+        }
+
+        serviceProvider().getUserInformation(newUsers, { onUsersInfoReceived(it, commentsMap) },
+            { Log.d(TAG, "Error when fetching users information") })
+    }
+
+    override fun onUsersInfoReceived(users: Map<String, User>, commentsMap: Map<String, Comment>) {
+        Log.d(TAG, "on users info received")
```

Comment:
Is this leftover from debugging?

Reply:
This was part is part of Marcel's code I have merged, so I don't know what it is used for

Reply:
Oh the Log, yeah sorry I will remove that